src/component/addons/AddonList/ConfiguredAddons/ConfiguredAddons.jsx

```diff
@@ -17,15 +17,51 @@ import {
 import { Link } from 'react-router-dom';
 import PageContent from '../../../common/PageContent/PageContent';
 import PropTypes from 'prop-types';
+import useAddons from '../../../../hooks/api/getters/useAddons/useAddons';
+import useToast from '../../../../hooks/useToast';
+import useAddonsApi from '../../../../hooks/api/actions/useAddonsApi/useAddonsApi';
+
+const ConfiguredAddons = ({ addons, hasAccess, getIcon }) => {
+    const { refetchAddons } = useAddons();
+    const { updateAddon, removeAddon } = useAddonsApi();
+    const { setToastData } = useToast();
+
+    const toggleAddon = async addon => {
+        try {
+            await updateAddon({ ...addon, enabled: !addon.enabled });
+            refetchAddons();
+            setToastData({
+                type: 'success',
+                title: 'Success',
+                text: 'Addon state switched successfully',
+            });
+        } catch (e) {
+            setToastData({
```

Comment:
You can use setToastApiError here, which will save you having to specify the type